sdk/python/lib/pulumi/runtime/resource.py

```diff
@@ -140,7 +139,8 @@ async def prepare_resource(res: 'Resource',
         for dep in deps:
             urn = await dep.urn.future()
             urns.add(urn)
-            dependencies.add(urn)
+            if urn:
```

Comment:
Something weird was going on here in the SDK before the change, why `Set[Optional[str]]` is being returned for URN-sets? DO we care about a `None` tracked in the URN sets? Probably not, need to see if can refactor.

Reply:
> DO we care about a `None` tracked in the URN sets?

I don't think so. Do you know why `None` was being added to the set?

Reply:
I've poked around and it sounds like Optional[str] sequence is passed to proto layer; it looks harmless I don't think there's any semantics attached to this, we can probably tighten to str.